Assets/MixedRealityToolkit/Interfaces/InputSystem/Handlers/IMixedRealityPointerHandler.cs

```diff
@@ -34,4 +34,20 @@ public interface IMixedRealityPointerHandler : IEventSystemHandler
         /// <param name="eventData"></param>
         void OnPointerClicked(MixedRealityPointerEventData eventData);
     }
+
+    /// <summary>
+    /// Type of pointer event. Matches the methods defined in <see cref="IMixedRealityPointerHandler"/>.
+    /// </summary>
+    public enum PointerEventType
+    {
+        Down,
+        Up,
+        Dragged,
+        Clicked
+    }
+
+    /// <summary>
+    /// Delegate type for handling pointer events. Subscribe via <see cref="IMixedRealityInputSystem.PointerEvent"/>.
+    /// </summary>
+    public delegate void PointerHandler(MixedRealityPointerEventData eventData, PointerEventType eventType);
```

Comment:
Did we ever look into somehow keeping just the single interface (i.e. the IMixedRealityPointerHandler) to also be the way that we get these non-Unity UI events? i.e. instead of storing a list of just GameObjects, potentially a list of IEventSystemHandler?

The reason I ask is it seems odd to have duplicate definitions of both a PointerEventType enum and an interface. This will probably set a precedent going forward and if the goal is to make this the way we really wanna split Unity events + c# events and have agreement then maybe it's fine, but if not I'm curious if there was some way to avoid this new concept...

Otherwise, we could always go the route of the default pointer mediator (which is fed information directly from the focus provider), which is also not a necessarily great design (in that it depends on a specific caller to give them the information, rather than generally requesting it as is done here).

Anyways, I guess my thoughts here are:

1) What your thoughts are on being able to keep some consistent interface (i.e. storing gameobjects + raw interfaces that aren't game objects, so that the null checking will work without having to cast to game objects).
2) If that doesn't work, thoughts on following the default mediator approach.
3) If that doesn't work, having it written down somewhere what the exact input dispatch flow is (i.e. so we can point to that in the future when we're creating new things in the future)

I would probably still like to hear @keveleigh's thoughts here - he has a lot more history than I do with this area.

Reply:
Thanks for that feedback! I've created a new PR for an alternative approach where we have methods in the input system to subscribe and unsubscribe pointer handlers as global listeners: #4605